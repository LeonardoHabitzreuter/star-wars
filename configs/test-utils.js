import * as theme from '../src/styles/theme'
import React from 'react'
import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  )
}

const customRender = (ui, options) => render(
  ui,
  { wrapper: AllTheProviders, ...options }
)

export * from '@testing-library/react'

export { customRender as render }
