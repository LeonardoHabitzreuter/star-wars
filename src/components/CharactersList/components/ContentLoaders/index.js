import React from 'react'
import PropTypes from 'prop-types'
import ContentLoader from 'react-content-loader'
import { colors } from '/styles/theme'
import { StyledCharacter } from '../CharacterCard/styles'

// eslint-disable-next-line react/prop-types
const Loader = ({ hide }) => (
  <StyledCharacter hide={hide} data-testid="characterCard" >
    <ContentLoader
      uniqueKey='unique'
      speed={2}
      width='100%'
      height={160}
      viewBox='0 0 600 160'
      backgroundColor={colors.lightGray}
      foregroundColor={colors.darkGray}
    >

      <rect x='100' y='5' rx='3' ry='3' width='400' height='35' />
      <rect x='0' y='70' rx='3' ry='3' width='200' height='20' />
      <rect x='400' y='70' rx='3' ry='3' width='600' height='20' />
      <rect x='0' y='120' rx='3' ry='3' width='200' height='20' />
      <rect x='400' y='120' rx='3' ry='3' width='600' height='20' />
    </ContentLoader>
  </StyledCharacter>
)

const ContentLoaders = ({ loading }) => (
  <>
    <Loader hide={!loading} />
    <Loader hide={!loading} />
    <Loader hide={!loading} />
  </>
)

ContentLoaders.propTypes = {
  loading: PropTypes.bool.isRequired
}

export default ContentLoaders
