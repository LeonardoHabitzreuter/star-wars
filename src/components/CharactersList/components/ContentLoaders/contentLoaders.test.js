import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import React from 'react'
import ContentLoaders from './index'
import { getAllByTestId } from '@testing-library/dom'
import { render } from 'test-utils'

describe('ContentLoaders component', () => {
  test('should render properly', () => {
    const { container } = render(<ContentLoaders loading />)

    expect(container.firstChild).toMatchSnapshot()
  })

  test('should render nothing when is not loading', () => {
    const { container } = render(<ContentLoaders loading={false} />)

    const cards = getAllByTestId(container, 'characterCard')

    cards.forEach(card => {
      expect(card).toHaveStyle('display: none')
    })
  })
})
