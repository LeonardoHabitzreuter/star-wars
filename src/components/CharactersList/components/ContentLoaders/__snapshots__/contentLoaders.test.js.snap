// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentLoaders component should render properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 8px;
  border-radius: 4px;
  border: solid 1px #fff;
  box-sizing: border-box;
  height: 150px;
  width: 100%;
  --cols: 3;
  max-width: calc(100% / var(--cols) - 20px);
  margin: 0 10px;
  margin-bottom: 16px;
  box-shadow: 2px 2px 2px 2px rgba(0,0,0,0.14);
}

@media (max-width:1024px) {
  .c0 {
    --cols: 2;
  }
}

@media (max-width:414px) {
  .c0 {
    --cols: 1;
    max-width: 100%;
    margin: 10px 0;
  }
}

<div
  class="sc-AxjAm c0"
  data-testid="characterCard"
>
  <svg
    aria-labelledby="unique-aria"
    height="160"
    role="img"
    viewBox="0 0 600 160"
    width="100%"
  >
    <title
      id="unique-aria"
    >
      Loading...
    </title>
    <rect
      clip-path="url(#unique-diff)"
      height="100%"
      role="presentation"
      style="fill: url(#unique-animated-diff);"
      width="100%"
      x="0"
      y="0"
    />
    <defs
      role="presentation"
    >
      <clippath
        id="unique-diff"
      >
        <rect
          height="35"
          rx="3"
          ry="3"
          width="400"
          x="100"
          y="5"
        />
        <rect
          height="20"
          rx="3"
          ry="3"
          width="200"
          x="0"
          y="70"
        />
        <rect
          height="20"
          rx="3"
          ry="3"
          width="600"
          x="400"
          y="70"
        />
        <rect
          height="20"
          rx="3"
          ry="3"
          width="200"
          x="0"
          y="120"
        />
        <rect
          height="20"
          rx="3"
          ry="3"
          width="600"
          x="400"
          y="120"
        />
      </clippath>
      <lineargradient
        id="unique-animated-diff"
      >
        <stop
          offset="0%"
          stop-color="#f3f3f3"
          stop-opacity="1"
        >
          <animate
            attributeName="offset"
            dur="2s"
            keyTimes="0; 0.25; 1"
            repeatCount="indefinite"
            values="-2; -2; 1"
          />
        </stop>
        <stop
          offset="50%"
          stop-color="#dbdbdb"
          stop-opacity="1"
        >
          <animate
            attributeName="offset"
            dur="2s"
            keyTimes="0; 0.25; 1"
            repeatCount="indefinite"
            values="-1; -1; 2"
          />
        </stop>
        <stop
          offset="100%"
          stop-color="#f3f3f3"
          stop-opacity="1"
        >
          <animate
            attributeName="offset"
            dur="2s"
            keyTimes="0; 0.25; 1"
            repeatCount="indefinite"
            values="0; 0; 3"
          />
        </stop>
      </lineargradient>
    </defs>
  </svg>
</div>
`;
